tmail-backend/webadmin/webadmin-rate-limit/src/main/java/com/linagora/tmail/webadmin/RateLimitPlanUserRoutes.java

```diff
@@ -0,0 +1,192 @@
+                .block();
+        };
+    }
+
+    private Route getUsersOfPlan() {
+        return (request, response) -> {
+            RateLimitingPlanId planId = extractPlanId(request);
+            planIdPreconditions(planId);
+            return Flux.from(planUserRepository.listUsers(planId))
+                .map(Username::asString)
+                .collectList()
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Route getPlanOfUser() {
+        return (request, response) -> {
+            Username username = extractUsername(request);
+            userPreconditions(username);
+            return Mono.from(planUserRepository.getPlanByUser(username))
+                .map(planId -> new RateLimitingPlanIdResponse(planId.value().toString()))
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .onErrorResume(RateLimitationPlanNotFoundException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.NOT_FOUND_404)
+                        .type(ErrorResponder.ErrorType.NOT_FOUND)
+                        .message(String.format("User %s does not have a plan", username.asString()))
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Route revokePlanOfUser() {
+        return (request, response) -> {
+            Username username = extractUsername(request);
+            userPreconditions(username);
+            return Mono.from(planUserRepository.revokePlan(username))
+                .then(Mono.just(Responses.returnNoContent(response)))
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Username extractUsername(Request request) {
+        return Username.of(request.params(USERNAME_PARAM));
+    }
+
+    private RateLimitingPlanId extractPlanId(Request request) {
+        return new RateLimitingPlanId(UUID.fromString(request.params(PLAN_ID_PARAM)));
```

Comment:
```suggestion
        return RateLimitingPlanId.parse(request.params(PLAN_ID_PARAM));
```

Encapsulation issue. IMO we should not leak the underlying id format and add a `parse` helper method for doing just that!